.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,175 @@
+description: 'Downloads latest build of OpenSearch, installs a plugin, executes and script and then starts OpenSearch on localhost:9200'
+
+inputs:
+  platform:
+    description: 'What platform is this action running on, e.g. "windows-latest" ? '
+    required: true
+
+  opensearch-version:
+    description: 'The version of OpenSearch that should be used, e.g "3.0.0"'
+    required: true
+
+  plugin-name:
+    description: 'The the name of the plugin to use, such as security-dashboards-plugin'
+    required: true
+
+  plugin-start-script:
+    description: 'The file name for the configuration script for the plugin such as install_demo_configurations'
+    required: true
+  # plugins/opensearch-security/tools/install_demo_configuration.sh
+
+  java-version:
+    description: 'The java version to be used, must be either 11 or 17'
+    required: true
+
+  plugin-zip:
+    description: 'The name of the zip file for the plugin hosted on docker-host i.e. security-plugin.zip '
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    # Configure longpath names if on Windows
+    - name: Enable Longpaths if on Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: git config --system core.longpaths true
+      shell: pwsh
+
+    # Downloads the min version of OpenSearch Core from the latest build
+    - name: Download OpenSearch Core for Linux
+      if: ${{ inputs.platform == 'ubuntu-latest' }}
+      run: |
+        wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+      shell: bash
+
+    - name: Download OpenSearch Core for Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: |
+        Invoke-WebRequest https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip -Outfile opensearch-${{ inputs.opensearch-version }}.zip
+        tar -xzf opensearch-${{ inputs.opensearch-version }}.zip
+        del opensearch-${{ inputs.opensearch-version }}.zip
+        Rename-Item opensearch-* opensearch
+      shell: pwsh
+
+    - name: Download OpenSearch Core for MacOS
+      if: ${{ inputs.platform == 'macos-latest' }}
+      run: |
+        curl https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+        mv opensearch-* opensearch
+      shell: bash
+
+    # Assemble the specified plugin
+    - name: Assemble the specified plugin for MacOS and Linux
+      if: ${{ inputs.platform != 'windows-latest' }}
+      run: ./gradlew clean assemble
```

Comment:
I still recommend having the build step happen outside this action, but speaking of gradle, if you use `gradle/gradle-build-action@v2` its already multiplatform safe - did this in the CI workflow 

https://github.com/scrawfor99/security/blob/adc668f1a4e403c9ca4f96672166d6c498404c75/.github/workflows/ci.yml#L75-L79

Reply:
Do you mind explaining why you do not want the build in the action? I am not sure I saw your initial thoughts on it sorry.

Reply:
This simplifies the inputs for this workflow to roughly, "given a plugin zip file, setup and deploy an instance of OpenSearch with that plugin installed for testing".

Since we use different JDKs versions this is extra complexity we can have the caller of the workflow deal with, rather than integrating here.

Reply:
Sounds good.